CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,111 @@
+
+The enforcers have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions on project-managed platforms that are not aligned to this Code of Conduct. They will communicate reasons for moderation decisions when appropriate. This Code of Conduct governs behavior within our community spaces; it does not extend to governing on-chain interactions which are subject to the rules of the protocol itself.
+
+## Scope
+
+This Code of Conduct applies within all project-managed community spaces, including but not limited to our GitHub repositories, official Discord server, community forums, Telegram groups, and during project-related events. It also applies when an individual is officially representing the community in public spaces. Examples of representing our community include using an official project email address, posting via an official social media account, or acting as an appointed representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the enforcers (as defined in "Enforcement Responsibilities") at
+[TODO: insert contact].
+All complaints will be reviewed and investigated promptly and fairly.
+
+All community leaders are obligated to respect the privacy and security of the
+reporter of any incident. We strive to apply this Code of Conduct consistently, but acknowledge that the permissionless and pseudonymous nature of some blockchain interactions may present challenges to enforcement beyond project-controlled platforms. Our focus is on maintaining a healthy environment within the community spaces we manage.
+
+## Enforcement Guidelines
+
+Community leaders will follow these Community Impact Guidelines in determining
+the consequences for any action they deem in violation of this Code of Conduct:
+
+### 1. Correction
+
+**Community Impact**: Use of inappropriate language or other behavior deemed
+unprofessional or unwelcome in the community.
+
+**Consequence**: A private, written warning from community leaders, providing
+clarity around the nature of the violation and an explanation of why the
+behavior was inappropriate. A public apology may be requested.
+
+### 2. Warning
+
+**Community Impact**: A violation through a single incident or series of
+actions.
+
+**Consequence**: A warning with consequences for continued behavior. No
+interaction with the people involved, including unsolicited interaction with
+those enforcing the Code of Conduct, for a specified period of time. This
+includes avoiding interactions in community spaces as well as external channels
+like social media. Violating these terms may lead to a temporary or permanent
+ban.
+
+### 3. Permanent Ban
```

Comment:
I removed temporary ban, I think given the nature of our industry this makes sense. If warnings are ignored, permanent bans is justified imo